src/Orleans/CodeGeneration/TypeUtils.cs

```diff
@@ -310,10 +310,20 @@ public static bool IsGrainClass(Type type)
             return !IsGeneratedType(type);
         }
 
-        public static bool IsSystemTargetClass(Type type)
+        internal static bool IsSystemTargetClass(Type type)
+        {
+            IEnumerable<string> violations;
+            return IsSystemTargetClass(type, out violations, false);
+        }
+
+        public static bool IsSystemTargetClass(Type type, out IEnumerable<string> violations, bool complain)
```

Comment:
It looks a bit strange to me that an `IsX()` method may return complains. Usually, an `IsX()` method only returns true or false. I think it would be better to have an explicit `ValidateSystemTargetClass()` method or something like that and not mix validation into `IsSystemTargetClass()`.